mesonbuild/backend/ninjabackend.py

```diff
@@ -2926,7 +2926,7 @@ def generate_single_compile(self, target: build.BuildTarget, src,
 
         # Include PCH header as first thing as it must be the first one or it will be
         # ignored by gcc https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100462
-        use_pch = self.target_uses_pch(target) and is_generated != 'pch'
+        use_pch = self.target_uses_pch(target)
```

Comment:
This code was originally added in 739e499554850d2cb8268bed219e8ba127051f76, and I cannot tell what it was supposed to do at the time...